code/modules/reagents/chemistry_reagents/medical.dm

```diff
@@ -43,6 +42,52 @@
 			if(prob(10))
 				M.emote(pick("twitch","blink_r","shiver"))
 
+
+/datum/reagent/hypervene
+	name = "Hypervene"
+	id = "hypervene"
+	description = "Quickly purges the body of toxin damage, radiation and all other chemicals. Causes significant pain."
+	reagent_state = LIQUID
+	color = "#19C832"
+	overdose = REAGENTS_OVERDOSE / 2
+	overdose_critical = REAGENTS_OVERDOSE_CRITICAL / 2
+	scannable = 1
+
+	on_mob_life(mob/living/M, alien)
+		. = ..()
+		if(!.) return
+		for(var/datum/reagent/R in M.reagents.reagent_list)
+			if(R != src)
+				M.reagents.remove_reagent(R.id,4 * REM)
+		M.reagent_shock_modifier += 50 //Significant pain while metabolized.
```

Comment:
Create new defines if necessary, and use stuff like PAIN_REDUCTION_LIGHT instead of numbers here. Magical numbers and all.